File: 11/1.rb
# You are given two sorted arrays, A and B, where A has a large enough buffer
# at the end to hold B. Write a method to merge B into A in sorted order

require_relative "merge_sort"

array1 = [9, 7, 4, 1, 2, 8, 6]
array2 = [10, 23, 62, 5, 3]

def merge_arrays(array1, array2)
  merge_sort array1 + array2
end

p merge_arrays(array1, array2)

File: 11/python/1.py
from merge_sort import merge, merge_sort

import random

def merge_arrays(array1, array2):
    return merge_sort(merge(array1, array2))

array_a = [random.randint(0, 100) for x in range(0, 10)]
array_b = [random.randint(0, 100) for x in range(0, 10)]

print merge_arrays(array_a, array_b)
